src/app/Dice.elm

```diff
@@ -0,0 +1,19 @@
+module Dice exposing (emptyCup, Cup, cupFromIntList)
+
+
+type alias Die =
+    Int
+
+
+type alias Cup =
+    List Die
+
+
+emptyCup : Cup
+emptyCup =
+    []
+
+
+cupFromIntList : List Int -> Cup
+cupFromIntList =
+    identity
```

Comment:
It's not a constructor, though. It's like `Dict.fromList`. Should that be called `Dict.init` ?

I'm not a huge fan of following rules for rules' sake. In this case, it's not a model. I'm actually *not* necessarily "following" TEA here, though there may be elements of TEA present... (because before TEA came along as a thing, composition and delegation was still just a very good pattern to follow).